Ny oppgave om lister og indekser.

okpedersen opened the pull request:
Introduserer `lst.append()`, `lst.remove()` og bruk av indekser på både lister og strenger.

okpedersen committed:
Ny oppgave om lister og indekser.

arve0 commented on src/python2/lister_og_indekser/lister_og_indekser.md:
````diff
@@ -0,0 +1,252 @@
+
+```python
+lst = ['Per', 'Kim', 'Ada']
+for i in range(len(lst))
````
enumerate?

okpedersen replied:
Jeg var inne på tanken, men jeg tror det er bedre å introdusere det generelle konseptet først, selv om det ikke er idiomatisk. Jeg tror det er bedre å introdusere det på et senere tidspunkt.

gahjelle replied:
Jeg er enig med Arve. Det generelle konseptet bak løkker i python i er iteratorer og `for element in lst:`. Det er aldri noen grunn til å skrive `for i in range(len(lst)):` i Python, og jeg ser derfor ikke helt hvorfor vi skal lære det bort? Indekstreningen hører nok bedre sammen med slicing enn med for-løkker. :)

okpedersen committed:
Endret range(len(lst)) til enumerate(lst), samt protips, oppgaver og
utfordringer deretter.

okpedersen commented:
Godt poeng @gahjelle, jeg tenkte mest på at indekser er det som brukes mest om man ser på C++ o.l. Oppgaven er nå endret til `enumerate()`.

gahjelle commented:
Man kunne eventuelt hatt med en liten boks som sier noe sånt som

### For-løkker i andre programmeringsspråk

I mange andre programmeringsspråk må man bruke indekser for å gå gjennom elementene i en liste ... blah blah ... For eksempel i Java vil man skrive`for (int i = 0; i < lst.length; i++) sum += lst[i]` ...

Men jeg vet ikke om det ville vært nyttig?

gahjelle commented on src/python2/lister_og_indekser/lister_og_indekser.md:
````diff
@@ -0,0 +1,227 @@
+
+```python
+>>> lst = ['Per', 'Ada', 'Kim']
+>>> lst.remove("Kim")
````
Vil `lst.remove('Ada')` være ørlite bedre? Da får vi med poenget om at du kan fjerne elementer inne i en liste, ikke bare på slutten ...

okpedersen replied:
Jeg ser nå at hele protipet er litt lite spesifikt, forklarer ikke eksplisitt hvordan `.append()` og `.remove()` fungerer, bare gir et eksempel. Skal skrive om.

okpedersen commented:
Jeg tror bare det hadde vært nyttig for de få som faktisk har programmert i Java e.l. før og forvirrende for resten.

gahjelle commented:
Enig :)

arve0 commented:
Ser dere har blitt enig, men slenger på en rettesnorer jeg trur er nyttig for oss:

Lær bort ting som er nyttig framfor forståelse av konsept. Altså, man trenger ikke lære om en spesiell teknikk hvis den ikke brukes til noe i en påfølgende eller samme oppgave.

okpedersen committed:
Skrev om deler av remove/append protip

arve0 committed:
revidert "Lister og indekser"

- Prøvd å gjøre setninger så konsis som mulig. Unngå feks "er kjempeviktig",
  som strengt tatt ikke er nødvendig informasjon.
- Prøvd å skape et sterkere skille mellom hva som er eksempler, og hva som er
  oppgave. Eks: Har brukt "skal" framfor "har lyst" osv.
- Stegene: Litt mer detaljert hva som kreves "programmeringsmessig", framfor
  hvordan programmet skal oppføre seg.

arve0 commented:
Revidert i https://github.com/okpedersen/oppgaver/pull/4

okpedersen committed:
Merge pull request #4 from arve0/vg_python_lister_1

revidert "Lister og indekser"

okpedersen commented:
Merget okpedersen#4, mye bra der.
